worker/leaseexpiry/worker.go

```diff
@@ -100,48 +99,36 @@ func (w *expiryWorker) loop() error {
 }
 
 func (w *expiryWorker) expireLeases(ctx context.Context) error {
-	err := w.trackedDB.DB(func(db *sql.DB) error {
-		// Txn here provides the transaction semantics we need, but there are
-		// no retry strategies employed. This is a special case, in this instance
-		// we will retry this in one second and other controllers in a HA
-		// setup will also retry.
-		return database.Txn(ctx, db, func(ctx context.Context, tx *sql.Tx) error {
-			res, err := tx.ExecContext(ctx, w.dml)
-			if err != nil {
-				// TODO (manadart 2022-12-15): This incarnation of the worker runs on
-				// all controller nodes. Retryable errors are those that occur due to
-				// locking or other contention. We know we will retry very soon,
-				// so just log and indicate success for these cases.
-				// Rethink this if the worker cardinality changes to be singular.
-				if txn.IsErrRetryable(err) {
-					w.logger.Debugf("ignoring error during lease expiry: %s", err.Error())
-					return nil
-				}
-				return errors.Trace(err)
+	err := w.trackedDB.TxnNoRetry(ctx, func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
The change can be seen here. the ergonomics of running without attempting retries is much nicer.